sound/soc/sof/topology.c

```diff
@@ -3046,8 +3046,25 @@ static void sof_complete(struct snd_soc_component *scomp)
 static int sof_manifest(struct snd_soc_component *scomp, int index,
 			struct snd_soc_tplg_manifest *man)
 {
-	/* not currently parsed */
-	return 0;
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
+
+	/* backward compltible with tplg without ABI info */
+	if (man->priv.size == 0)
```

Comment:
if (!man->priv.size)

Reply:
@bardliao on second thoughts, with older topologies that have no ABI info, we'd never get here isnt it? There should be no manifest section.

Reply:
@ranj063 There is always manifest section which contains number of each type elements
https://github.com/torvalds/linux/blob/master/include/uapi/sound/asoc.h#L368
And our topology is to add ABI information in its private data field.

Reply:
@bardliao I just built the topology with an older commit and there is not SectionManifest in the conf file. What am I missing?

Reply:
@ranj063 You won't see it in the conf file. I think it is generated by topology itself since it needs to calculate the number of each type elements. You will see it in the conf file only when we add something in the private data.

Reply:
@bardliao thanks, that makes sense.